### DISCUSSION TITLE
How to use next/image Image component in Storybook or Jest?

### DISCUSSION THREAD

**Discussion by weyert** [vercel/next.js#18516] [Help]
I have written simple component which uses the new `Image`-component only when I try to render this simple component it's complaining about image configuration. I was wondering what the right approach is to render it, also it appears that using `Image`-component in a storybook story is also giving errors.

Error when rendering in Jest using Testing Library: `TypeError: Cannot destructure property 'deviceSizes' of 'imageData' as it is undefined.` looking at the good it's reading these configuration a environment variable so I have something similar to the code snippet posted below:

```javascript
  const originalEnv = process.env

  beforeAll(() => {
    // Setup the injected process environment
    process.env.__NEXT_IMAGE_OPTS = JSON.stringify({
    deviceSizes: [320, 420, 768, 1024, 1200],
    iconSizes: [16, 32, 48],
    })
  })

  afterEach(() => {
    // Reset the process environment variables
    process.env = originalEnv
  })
```

While when I try to render this same component in storybook I am getting errors like:

> Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
>
> Check the render method of `WidgetCard`.

How can I make this work? Do I need to mock away `next/image`? 
A mock like this appears to work but not sure if that's a good solution:

```
jest.mock('next/image', () => ({
  // eslint-disable-next-line @typescript-eslint/naming-convention
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line jsx-a11y/alt-text
    return <img {...props} />
  },
}))
```

**Comment by aippili-asp** [comment]
Were you able to solve the issue?

**Reply by weyert** [reply]
I have been able to get it working in Jest but not in Storybook yet but there are some PRs here and in the `stortybook `project to assist with this.

**Reply by robbertkl** [reply]
Could you link these? Would love to track.

**Comment by leerob** [comment]
FYI, we are tracking this here.

https://github.com/vercel/next.js/issues/18393
